static/src/javascripts/projects/commercial/modules/sticky-top-banner.js

```diff
@@ -0,0 +1,210 @@
+    if (!config.page.hasSuperStickyBanner) {
+        addEventListener(window, 'scroll', onScroll, {
+            passive: true,
+        });
+    }
+};
+
+const getAdvertSizeByIndex = (advert: ?Advert, index: number): ?number => {
+    if (advert && advert.size && typeof advert.size !== 'string') {
+        return advert.size[index];
+    }
+};
+
+const onFirstRender = (): Promise<any> =>
+    trackAdRender(topSlotId).then(isRendered => {
+        if (isRendered) {
+            const advert = getAdvertById(topSlotId);
+            const adSize0 = getAdvertSizeByIndex(advert, 0);
+            const adSize1 = getAdvertSizeByIndex(advert, 1);
+
+            if (
+                // skip for Fabric creatives
+                adSize0 !== 88 &&
+                // skip for native ads
+                adSize1 &&
+                adSize1 > 0
+            ) {
+                return fastdom
+                    .read(() => {
+                        const styles = window.getComputedStyle(topSlot);
+
+                        return (
+                            parseInt(styles.paddingTop, 10) +
+                                parseInt(styles.paddingBottom, 10) +
+                                adSize1 || 0
+                        );
+                    })
+                    .then(resizeStickyBanner);
+            }
+            return fastdom
+                .read(() => {
+                    if (topSlot) {
+                        return topSlot.offsetHeight;
+                    }
+
+                    return 0;
+                })
+                .then(resizeStickyBanner);
+        }
+    });
+
+const initState = (): Promise<any> =>
+    fastdom
+        .read(() => {
+            if (header) {
+                headerHeight =
+                    parseInt(window.getComputedStyle(header).height, 10) || 0;
+            }
+            if (topSlot) {
+                return (
+                    parseInt(window.getComputedStyle(topSlot).height, 10) || 0
+                );
+            }
+
+            return 0;
+        })
+        .then(currentHeight =>
+            Promise.all([resizeStickyBanner(currentHeight), onScroll()])
+        );
+
+const initStickyTopBanner = (): Promise<void> => {
+    if (!commercialFeatures.stickyTopBannerAd) {
+        return Promise.resolve();
+    }
+
+    topSlot = document.getElementById(topSlotId);
+    if (
+        topSlot &&
+        detect.isBreakpoint({
+            min: 'desktop',
+        })
+    ) {
+        header = document.getElementById('header');
+        stickyBanner = ((topSlot.parentNode: any): HTMLElement);
+
+        // First, let's assign some default values so that everything
+        // is in good order before we start animating changes in height
+        return (
+            initState()
+                // Second, start listening for height and scroll changes
+                .then(setupListeners)
+                .then(onFirstRender)
+        );
+    }
+    topSlot = null;
```

Comment:
It can be defined and the breakpoint is lower than desktop. That is to prevent memory leaks.